pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -184,6 +216,7 @@ func NewForSchemas(ctx context.Context, client kube.Client, revision, domainSuff
 			if err != nil {
 				return nil, err
 			}
+			out.queueMap[s.Resource().GroupVersionKind()] = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), s.Resource().GroupVersionKind().String())
```

Comment:
Why do we need a queue per type?

Reply:
As above, to make it faster instead of accumulating

Reply:
Does all of our downstream usages support concurrent calls?

TBH my impression is the event handlers are extremely fast. I would not support adding complexity without proof concurrency will make it faster. It doesn't intuitively make sense to me

Reply:
>Does all of our downstream usages support concurrent calls?

Need to have a look

>TBH my impression is the event handlers are extremely fast. I would not support adding complexity without proof concurrency will make it faster. It doesn't intuitively make sense to me

It is fast, but the producer is from k8s informer event callback which is faster and from multi go routines.